pkg/kubelet/kubelet_node_status.go

```diff
@@ -91,7 +93,8 @@ func (kl *Kubelet) tryRegisterWithAPIServer(node *v1.Node) bool {
 		return true
 	}
 
-	if !apierrors.IsAlreadyExists(err) {
+	continueOnForbidden := !utilfeature.DefaultFeatureGate.Enabled(features.DisableKubeletRegisterGetOnForbidden)
```

Comment:
so use cases here:

- create forbidden, patch allowed, node exists: everything will work fine as expected.
- create forbidden, patch allowed, node not exists: we will return false from a bit later in code. Almost no change with today's behavior.
- both forbidden: we will return false from a bit later in code. So it's still OK. Almost no change with today's behavior.

finally we can always disable this behavior.

Reply:
Yup, that covers all the cases. Thanks!

Reply:
I posted it for other reviewers to simplify the review